docs/technical/pull-requests.md

```diff
@@ -0,0 +1,44 @@
+# Checking out pull requests from a forked repository
+PR [#3602](https://github.com/desktop/desktop/pull/3602) introduced the ability to checkout a branch from a forked repository. In order to accomplish this, we needed a way to manage remotes on your behalf. This document is intended to detail the process we developed to make checking out PRs as frictionless as possible.
+
+## Magic Remote Prefix
+One of the main problems we needed to solve was determining which remotes are no longer needed and can be cleaned. We decided to prefix the remotes we add on your behalf with a magic string: `github-desktop-`
+
+```ts
+export const ForkedRemotePrefix = 'github-desktop-'
+```
+[Code](https://github.com/desktop/desktop/blob/34a05b155ff69bb19cc4da5b2caa89856e3e63fb/app/src/lib/stores/pull-request-store.ts#L26)
+
+## Cleaning Remotes
+One of the goals of our design was to ensure that we don’t cause your remotes — `.git/refs/remotes` — to grow unbounded. We prevent this by cleaning up after ourselves. We determined that a remote is a candidate for removal when it meets the following conditions.
+
+* Start with our prefix
+* The PR associated with the remote is closed
+
+```ts
+private forkedRemotesToDelete(
```

Comment:
I worry that including a c&p of the code here means it'll fall out of sync with the _actual_ code we use. Could we describe the algorithm in some pseudo code, so we don't get bogged down in the implementation details?